app/controllers/goals_controller.rb

```diff
@@ -0,0 +1,21 @@
+class GoalsController < ApplicationController
+  def create
+    @user = current_user
+    @goal = @user.goals.build(goal_params)
+
+    if @goal.save!
```

Comment:
`@goal.save!` will raise an error if the goal is invalid. Consider using `@goal.save` (no bang) instead as it will return `false` if the goal is invalid which will allow the `else` block to run.